packages/vm/tests/api/runTx.js

```diff
@@ -84,11 +84,28 @@ tape('should fail when account balance overflows', async t => {
   await suite.putAccount(tx.from.toString('hex'), from)
   await suite.putAccount(tx.to, to)
 
-  shouldFail(t,
-    suite.runTx({ tx }),
-    (e) => t.equal(e.message, 'Value overflow')
-  )
+  const res = await suite.runTx({ tx })
+
+  t.equal(res.execResult.exceptionError.error, 'Value overflow')
+  t.equal(vm.stateManager._checkpointCount, 0)
+  t.end()
+})
+
+tape('should fail when account balance overflows (create)', async t => {
+  const vm = new VM()
+  const suite = setup(vm)
 
+  const contractAddress = Buffer.from('37d6c3cdbc9304cad74eef8e18a85ed54263b4e7', 'hex')
+  const tx = getTransaction(true, true, '0x01', true)
+  const from = createAccount()
+  const to = createAccount('0x00', ethUtil.MAX_INTEGER)
+  await suite.putAccount(tx.from.toString('hex'), from)
+  await suite.putAccount(contractAddress, to)
+
+  const res = await suite.runTx({ tx })
+
+  t.equal(res.execResult.exceptionError.error, 'Value overflow')
+  t.equal(vm.stateManager._checkpointCount, 0)
```

Comment:
Right, performing the assert in checkpoint count ensures a clean state afterwards. 👍